common/nmea_parser.h

```diff
@@ -0,0 +1,213 @@
+    units::length::meter_t altitude;                       // Altitude above ellipsoid
+    units::velocity::meters_per_second_t velocity;         // velocity
+    int numberOfSatelites;                                 // Currently observed number of satelites
+    double horizontalDilution;                             // horizontal dilution - lower value is better
+    GPSQuality gpsQuality;                                 // GPS quality indicator
+    TimeStamp time;                                        // time of measurement
+};
+
+class NMEAParser {
+
+    using second_t = units::time::second_t;
+    using minute_t = units::time::minute_t;
+    using hour_t = units::time::hour_t;
+    using millisecond_t = units::time::millisecond_t;
+    using meter_t = units::length::meter_t;
+    using degree_t = units::angle::degree_t;
+    using arcminute_t = units::angle::arcminute_t;
+    using meters_per_second_t = units::velocity::meters_per_second_t;
+
+    private:
+
+    static std::vector<std::string> parseNMEAstring(const std::string &textToParse, const std::string &NMEA_sentence_id) {
+
+        using namespace std;
+        const char delimiter = '$';      // sentences separated by [$]
+        const char w_delimiter = ',';    // elements separated by  [,]
+
+        vector<string> sentences,words;
+        istringstream split(textToParse);
+        // separating the text to sentences
+        try {
+            if (textToParse == "") throw "Empty string to parse";
+
+            for (string each; getline(split, each, delimiter); sentences.push_back(each));    
+            // find the sentence with the id we want
+            int sentenceNumber= -1;
+            for (vector<int>::size_type i = 0; i < sentences.size(); i++) {
+                size_t found = sentences[i].find(NMEA_sentence_id);
+                if (found!= string::npos) {
+                    sentenceNumber = i;
+                    break;
+                }
+            }
+            if (sentenceNumber < 0) {
+                throw "[NMEAParser:parseNMEAstring()] : cannot find NMEA id";
+            }
+            auto f = sentences[sentenceNumber];
+            // separating the sentence to words 
+            istringstream splitWord(f);
+            for (string word; getline(splitWord, word, w_delimiter); words.push_back(word));
+
+        } catch(const char *error) {
+            throw error;
+        } catch(...) {}
+
+        return words;
+    }
+
+    static GPSQuality getGpsQuality(const int &qualityId) {     
+        GPSQuality quality;  
+        switch(qualityId) {
+            case 0 : quality = GPSQuality::INVALID; break;
+            case 1 : quality = GPSQuality::GPSFIX; break;
+            case 2 : quality = GPSQuality::DGPSFIX; break;
+            case 3 : quality = GPSQuality::PPSFIX; break;
+            case 4 : quality = GPSQuality::RTK; break;
+            case 5 : quality = GPSQuality::FRTK; break;
+        }
```

Comment:
I'd drop this function and just cast to a ``GPSQuality`` (see comment above)

Reply:
:+1: